Blog for v1 release - for 11 July (#910)
kubevirt/kubevirt.github.io: _posts/2023-07-11-KubeVirt-v1-has-landed.md

---
layout: post
author: KubeVirt Maintainers
title: KubeVirt v1.0 has landed!
description: We are very pleased to announce the release of KubeVirt v1.0!
navbar_active: 
pub-date: July 11
pug-year: 2023
category: news
tags:
  [
    "KubeVirt",
    "v1.0",
    "release",
    "community",
    "cncf",
    "milestone",
    "party time"
  ]

---

The KubeVirt community is proud to announce the release of KubeVirt v1.0! This release demonstrates the accomplishments of the community and user adoption over the years and represents an important milestone for everyone involved.

## A brief history

> The KubeVirt project started in Red Hat at the end of 2016, with the question: Can virtual machines (VMs) run in containers and be deployed by Kubernetes? When it proved not only possible, but also as a robust solution to the future of virtual machines in the container age, KubeVirt joined the [CNCF](https://www.cncf.io/) as a Sandbox project in September 2019, and an Incubating project in April 2022. From a handful of people hacking away on a proof of concept, KubeVirt has grown into 45 active repositories, with the primary [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt) repo having 17k commits and almost 1k forks.

Reviewer: That's a couple really long sentences; mind if I offer a breaking-up?

```suggestion
The KubeVirt project started in Red Hat at the end of 2016, with the question: Can virtual machines (VMs) run in containers and be deployed by Kubernetes?
It soon  proved possible, and beyond that, displayed promise as a robust solution to the future of virtual machines in the container age.
KubeVirt joined the [CNCF](https://www.cncf.io/) as a Sandbox project in September 2019, and graduated to Incubating in April 2022.
From a handful of people hacking away on a proof of concept, KubeVirt has grown into a community with 45 active repositories. The primary [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt) repo sums 17k commits and almost 1k forks.
```

Hack that as you'd like -- I'm not 100% happy with my draft either. I tried to keep active voice at least.

Author: Thanks for spotting that. I've gone with "It proved to be not only possible, but quickly emerged as a promising solution to the future of virtual machines in the container age."

Leading up to this release we had a shift in release cadence: from monthly to 3 times a year, following the Kubernetes release model. This allows our developer community additional time to ensure stability and compatibility, our users more time to plan and comfortably upgrade, and also aligns our releases with Kubernetes to simplify maintenance and supportability.

The theme 'aligning with Kubernetes' is also felt through the other parts of the community, by following their governance processes; introducing SIGs to split test and review responsibilities, as well as a SIG release repo to handle everything related to a release; and regular [SIG meetings](https://calendar.google.com/calendar/u/0/embed?src=kubevirt@cncf.io) that now include SIG scale and performance and SIG storage alongside our weekly Community meetings.

## What’s included in this release?

This release demonstrates the accomplishments of the community and user adoption over the past many months. The full list of feature and bug fixes can be found in our [release notes](https://github.com/kubevirt/kubevirt/releases/tag/v1.0.0), but we’ve also asked representatives from some of our SIGs for a summary.

> ### SIG scale and performance

Reviewer 2: @alaypatel07 cc.

[Performance and scalability graphs from 0.58 to v1.0](https://github.com/kubevirt/kubevirt/blob/main/docs/release-v1-perf-scale-benchmarks.md)

> KubeVirt’s SIG-scale is focused on driving performance and scalability initiatives in the community. Our focus for the v1 release was on sharing the performance results over the past 3 releases - v0.58 to v0.59 to v1.

Reviewer: ```suggestion
KubeVirt’s SIG-scale is focused on driving performance and scalability initiatives in the community. Our theme for the v1 release was on sharing the performance results over the past 3 releases - v0.58 to v0.59 to v1.
```
Just an idea to avoid saying focus twice

Author: Kept that focus but changed the preceding sentence to "KubeVirt’s SIG-scale drives the performance and scalability initiatives in the community. Our focus for ..."

> By viewing these measurements over releases, it highlights the change in performance and scalability over time so the community and end-users know what to expect. The changes also highlight the effects each code change has so community members can diagnose.

Reviewer: ```suggestion
Viewing these measurements over multiple releases highlights the change in performance and scalability over time so the community and end-users know what to expect. The results highlight the effects each code change has so community members can diagnose.
```

Author: Great suggestion, thank you

End-users can use the same tools and techniques SIG-scale uses to analyze their own performance and scalability. Since performance and scalability are mostly relative, analyzing production KubeVirt deployments will be important to further contextualize the community's measurements.

[Performance benchmarks for v1.0 release](https://github.com/kubevirt/kubevirt/blob/main/docs/release-v1-perf-scale-benchmarks.md#performance-benchmarks-for-v1-release)

[Scalability benchmarks for v1.0 release](https://github.com/kubevirt/kubevirt/blob/main/docs/release-v1-perf-scale-benchmarks.md#scalability-benchmarks-for-v1-release)

### SIG storage
SIG-storage is focused on providing persistent storage to KubeVirt VMs and managing that storage throughout the lifecycle of the VM. This begins with provisioning and populating PVCs with bootable images but also includes features such as disk hotplug, snapshots, backup and restore, disaster recovery, and virtual machine export.

> For v1.0, SIG-storage has been focused on the following features: providing a flexible VM export API, enabling persistent SCSI reservation, provisioning VMs from a retained snapshot, and setting out-of-the-box defaults for additional storage provisioners. Another major effort was to implement Volume Populator alternatives to the KubeVirt DataVolume API in order to better leverage platform capabilities. The SIG meets every 2 weeks and welcomes anyone to join us for interesting storage-focussed discussions.

Reviewer: ```suggestion
For v1.0, SIG-storage has been focused on the following features: providing a flexible VM export API, enabling persistent SCSI reservation, provisioning VMs from a retained snapshot, and setting out-of-the-box defaults for additional storage provisioners. Another major effort was to implement Volume Populator alternatives to the KubeVirt DataVolume API in order to better leverage platform capabilities. The SIG meets every 2 weeks and welcomes anyone to join us for interesting storage-focused discussions.
```

Author: In light of the proliferation of other instances of 'focus', I dropped it in favour  of 'interesting storage discussions'

### SIG compute
SIG-compute is focused on the core virtualization functionality of KubeVirt, but also encompasses features that don’t fit well into another SIG. Some examples of SIG-compute’s scope include the lifecycle of VMs, migration, as well as maintenance of the core API.

For 1.0, SIG-compute has been focused on features for memory over-commit. This includes initial support for KSM and FreePageReporting. We added support for persistent vTPM, which makes it much easier to use BitLocker on Windows installs. Additionally, there's now an initial implementation for CPU Hotplug (currently hidden behind a feature gate).

### SIG network
The focus of SIG-network is to enhance and maintain all aspects of Virtual Machine network connectivity and management in KubeVirt.

> For the v1.0 release, we have introduced HotPlug and HotUnplug (as Alpha), which enables users to add and remove VM secondary network interfaces that use bridge binding on a running VM. HotPlug API stabilization and support for SR-IOV interfaces is under work for the next minor release.

Reviewer: ```suggestion
For the v1.0 release, we have introduced HotPlug and HotUnplug (as Alpha), which enables users to add and remove VM secondary network interfaces that use bridge binding on a running VM. HotPlug API stabilization and support for SR-IOV interfaces is under development for the next minor release.
```

### SIG-compute
SIG-compute is focused on the core virtualization functionality of KubeVirt, but also encompasses features that don’t fit well into another SIG. Some examples of SIG-compute’s scope include the lifecycle of VMs, migration, as well as maintenance of the core API.

For v1.0, SIG-compute developed features for memory over-commit. This includes initial support for KSM and FreePageReporting. We added support for persistent vTPM, which makes it much easier to use BitLocker on Windows installs. Additionally, there's now an initial implementation for CPU Hotplug (currently hidden behind a feature gate).

### SIG-network
SIG-network is committed to enhancing and maintaining all aspects of Virtual Machine network connectivity and management in KubeVirt.

> For the v1.0 release, we have introduced HotPlug and HotUnplug (as Alpha), which enables users to add and remove VM secondary network interfaces that use bridge binding on a running VM. HotPlug API stabilization and support for SR-IOV interfaces is under development for the next minor release.

Reviewer 3: ```suggestion
For the v1.0 release, we have introduced hot plug and hot unplug (as alpha), which enables users to add and remove VM secondary network interfaces that use bridge binding on a running VM. Hot plug API stabilization and support for SR-IOV interfaces is under development for the next minor release.
```

The theme 'aligning with Kubernetes' is also felt through the other parts of the community, by following their governance processes; introducing SIGs to split test and review responsibilities, as well as a SIG release repo to handle everything related to a release; and regular [SIG meetings](https://calendar.google.com/calendar/u/0/embed?src=kubevirt@cncf.io) that now include SIG scale and performance and SIG storage alongside our weekly Community meetings.

## What’s included in this release?

This release demonstrates the accomplishments of the community and user adoption over the past many months. The full list of feature and bug fixes can be found in our [release notes](https://github.com/kubevirt/kubevirt/releases/tag/v1.0.0), but we’ve also asked representatives from some of our SIGs for a summary.

### SIG-scale
[Performance and scalability graphs from 0.58 to v1.0](https://github.com/kubevirt/kubevirt/blob/main/docs/release-v1-perf-scale-benchmarks.md)

> KubeVirt’s SIG-scale drives the performance and scalability initiatives in the community. Our focus for the v1.0 release was on sharing the performance results over the past 3 releases - v0.58 to v0.59 to v1.0.

Author: Makes sense. Let me know if having these in parentheses is too cumbersome.

> Viewing these measurements over multiple releases highlights the change in performance and scalability over time so that the community and end-users know what to expect. The results also help to identify the effects each code change has so community members can diagnose.

Reviewer 4: Suggest modifying last sentence as follows
```
The results also help to identify the effects each code change has so community members can diagnose performance problems and regressions.
```

Author: Updated more or less as suggested (added some 'that's)

End-users can use the same tools and techniques SIG-scale uses to analyze their own performance and scalability. Since performance and scalability are mostly relative, analyzing production KubeVirt deployments will be important to further contextualize the community's measurements.

> [Performance benchmarks for v1.0 release](https://github.com/kubevirt/kubevirt/blob/main/docs/release-v1-perf-scale-benchmarks.md#performance-benchmarks-for-v1-release)

Reviewer 2: Point to v1-release branch.  The MR will get backported once it merges into main.

### SIG-scale

> [Performance and scalability graphs from 0.58 to v1.0](https://github.com/kubevirt/kubevirt/blob/main/docs/release-v1-perf-scale-benchmarks.md)

Reviewer 4: This line is not needed since we have the scalability and performance benchmarks links below after a short introduction

Author: Thanks for the latest round of changes. I've included them and dropped this ^ line.
